visualizers/visualization-common.js

```diff
@@ -0,0 +1,20 @@
+(function () {
+    document.addEventListener("keydown", function (e) {
+        if (e.key == " " || e.key == "Escape")
+            e.stopPropagation();
+            e.preventDefault();
+            window.frameElement.parentNode.dispatchEvent(new e.constructor(e.type, e))
+    });
+
+    document.addEventListener("keyup", function (e) {
+        if ((!e.ctrlKey && e.key == " ") || e.key == "Escape")
+            e.stopPropagation();
```

Comment:
I feel like it would be nice to create a function for this, like:
```javascript
function emitFromParent(event) {
    event.stopPropagation();
    event.preventDefault();
    window.frameElement.parentNode.dispatchEvent(new event.constructor(event.type, event));
}
```
It's up to you, though